src/CoreWCF.Http/tests/ServiceWithCoreWCFFaultContractTest.cs

```diff
@@ -0,0 +1,125 @@
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.IServiceWithCoreWCFFaultContract>(httpBinding,
+                    new System.ServiceModel.EndpointAddress(new Uri("http://localhost:8080/BasicWcfService/Service.svc")));
+                IServiceWithCoreWCFFaultContract channel = factory.CreateChannel();
+
+                var e = Assert.Throws<System.ServiceModel.FaultException<ClientContract.SSMCompatibilityFault>>(() => channel.Identity("test"));
+
+                ((IChannel)channel).Close();
+            }
+        }
+
+        [Fact]
+        [UseCulture("en-US")]
+        public async Task BasicScenarioServiceWithCoreWCFFaultContractWithHttpClient()
+        {
+            var client = _factory.CreateClient();
+            const string action = "http://tempuri.org/IServiceWithCoreWCFFaultContract/Identity";
+
+            var request = new HttpRequestMessage(HttpMethod.Post, new Uri("http://localhost:8080/BasicWcfService/Service.svc", UriKind.Absolute));
+            request.Headers.TryAddWithoutValidation("SOAPAction", $"\"{action}\"");
+
+            const string requestBody = @"<s:Envelope xmlns:s=""http://schemas.xmlsoap.org/soap/envelope/"" xmlns:tem=""http://tempuri.org/"">
+   <s:Header/>
+   <s:Body>
+      <tem:Identity>
+         <tem:Input>A</tem:Input>
+      </tem:Identity>
+   </s:Body>
+</s:Envelope>";
+
+            request.Content = new StringContent(requestBody, Encoding.UTF8, "text/xml");
+
+            //// FIXME: Commenting out this line will induce a chunked response, which will break the pre-read message parser
+            request.Content.Headers.ContentLength = Encoding.UTF8.GetByteCount(requestBody);
+
+            var response = await client.SendAsync(request);
+            Assert.False(response.IsSuccessStatusCode);
+
+            var responseBody = await response.Content.ReadAsStringAsync();
+            //_output.WriteLine(responseBody);
+
+            string expected = "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\">" +
+                "<s:Body><s:Fault>"
+                +"<faultcode>s:Client</faultcode>"
+                +"<faultstring"
+                + $"{GetXmlLangAttributeOrNot()}"
+                + ">The creator of this fault did not specify a Reason.</faultstring>"
+                +"<detail>"
+                +"<SSMCompatibilityFault xmlns=\"https://ssm-fault-contract-compatibility.com\" xmlns:a=\"http://schemas.datacontract.org/2004/07/Services\" xmlns:i=\"http://www.w3.org/2001/XMLSchema-instance\">"
+                +"<a:Message>An error occured</a:Message></SSMCompatibilityFault></detail></s:Fault></s:Body></s:Envelope>";
+
+            Assert.Equal(expected, responseBody);
+
+            string GetXmlLangAttributeOrNot()
+            {
+                const string enUsXmlLang = " xml:lang=\"en-US\"";
+#if NETCOREAPP3_1_OR_GREATER
+                return RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ?
```

Comment:
Did you find the response is different on Linux than on Windows? I wouldn't expect this to be the case. It might be a bug in the runtime. Can you open an issue for this and I can look into it at a later time.

Reply:
I opened #490

Reply:
Just noticed your earlier comment. I tracked down the problem, and it's caused by CultureInfo.CurrentCulture.Name returning String.Empty on Linux when we initialize the XmlLang property of FaultReasonText. This test is going to be flakey on non en-US systems and will fail for some people running tests locally. Instead can I suggest you fetch CultureInfo.CurrentCulture.Name and if it's length is greater then 0, then you add the xml:lang attribute with it's value? This will accommodate the different Linux behavior as well as make this test work for everyone.

Reply:
I’m running fr-FR culture thus I forced the culture to be en-US with `[UseCulture]` attribute on test method to make the culture name invariant.